ui/components/app/snaps/snap-ui-renderer/components/form.ts

```diff
@@ -0,0 +1,18 @@
+import { Form } from '@metamask/snaps-sdk';
+import { mapToTemplate } from '../utils';
+import { UIComponentFactory } from './types';
+
+export const form: UIComponentFactory<Form> = ({ element, ...params }) => ({
+  element: 'SnapUIForm',
+  // @ts-expect-error This is a problem with the types generated in the snaps repo.
+  children: element.children.map((children) =>
```

Comment:
If this is a typing issue that is not easy to solve, what are the arguments against using `(children: object)` or `(children: any)` and using `// @ts-expect-error` instead?

To me it looks like the `// @ts-expect-error` should be used only when there are no other solutions, it's kinda the last thing to do. I acknowledge that using `any` or `object` is bad practice, but `// @ts-expect-error` also looks even more bad IMO. WDYT?

Reply:
I was thinking we remove these once the TypeScript problems are solved in the snaps repo instead. If there is no comment it feels like more likely that we forget about it?

Reply:
Yes I would prefer leaving a `ts-expect-error` for things we want to resolve later

Reply:
```
If there is no comment it feels like more likely that we forget about it?
```
It can also be ``// TODO: ...```. But, if you prefer keeping `@ts-expect-error` temporarily, I'm alright with that.

Reply:
I will mark this as resolved. Answered here what's my opinion on such stuff: https://github.com/MetaMask/metamask-extension/pull/22828#discussion_r1496308545